Fix golint

## What happens?

The golint command is failing in Travis CI, but the tests are still passing

## What were you expecting to happen?

Travis CI should fail the build if there are any golint errors.

## Steps to reproduce: 

Run `golint -set_exit_status $(go list ./... | grep -v /vendor/); echo $?`

## Any errors, stacktrace, logs? 

Output from golint.
```
github.com/KohlsTechnology/git2consul-go/kv/handler.go:30:6: type name will be used as kv.KVHandler by other packages, and that stutters; consider calling this Handler
github.com/KohlsTechnology/git2consul-go/kv/mocks/kv.go:31:6: exported type KV should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/kv.go:36:1: exported method KV.Get should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/kv.go:44:1: exported method KV.Put should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/kv.go:53:1: exported method KV.Delete should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/kv.go:58:1: exported method KV.Txn should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:30:6: exported type Repo should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:39:1: exported method Repo.Name should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:43:1: exported method Repo.GetConfig should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:47:1: exported method Repo.Add should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:51:1: exported method Repo.CheckRef should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:55:1: exported method Repo.CheckoutBranch should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:60:1: exported method Repo.DiffStatus should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:69:1: exported method Repo.Head should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:77:1: exported method Repo.Pull should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:91:1: exported method Repo.ResolveRevision should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:96:1: exported method Repo.Worktree should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:100:1: exported method Repo.Lock should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:101:1: exported method Repo.Unlock should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:103:1: exported method Repo.GetStorer should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/kv/mocks/repository.go:107:1: exported method Repo.Branch should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/repository/mocks/repository.go:30:1: exported function InitRemote should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/repository/mocks/repository.go:48:1: exported function Add should have comment or be unexported
github.com/KohlsTechnology/git2consul-go/repository/mocks/repository.go:85:1: exported function Commit should have comment or be unexported
Found 24 lint suggestions; failing.
```

## Environment:

* Go 1.12
* Desktop OS/version: N/A

## Additional comments:

Need to add the `-set_exit_status` option for golint in the Travis config file to fix this.

In #3 I was able to fix all but one `golint` error ...

```
github.com/KohlsTechnology/git2consul-go/kv/handler.go:30:6: type name will be used as kv.KVHandler by other packages, and that stutters; consider calling this Handler
Found 1 lint suggestions; failing.
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+coverage.txt
 git2consul
 .idea
 *.swp

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,4 @@
-/build/bin
+git2consul
 .idea
 *.swp
 .vscode

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,19 +1,18 @@
 ---
+branches:
+  only:
+  - master
 language: go
 matrix:
   include:
     - go: 1.13.x
 
 before_install:
-  - export GO111MODULE=on
-  - go get golang.org/x/lint/golint
+  - GO111MODULE=off go get golang.org/x/lint/golint
 
 install:
   - # skip
 
 script:
-  - test -z $(gofmt -l .)
-  - golint $(go list ./...)
-  - go vet $(go list ./...)
-  - go test $(go list ./...)
-  - go build
+  - make test-dirty
+  - make test

--- a/Makefile
+++ b/Makefile
@@ -1,0 +1,37 @@
+BINARY = git2consul
+
+.PHONY: all
+all: build
+
+.PHONY: clean
+clean:
+	rm -rf $(BINARY)
+
+.PHONY: build
+build:
+	CGO_ENABLED=0 go build -o $(BINARY)
+
+.PHONY: test
+test: fmt lint vet test-unit
+
+.PHONY: test-unit
+test-unit:
+	go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+
+# Make sure go.mod and go.sum are not modified
+.PHONY: test-dirty
+test-dirty: build
+	go mod tidy
+	git diff --exit-code
+
+.PHONY: fmt
+fmt:
+	test -z "$(shell gofmt -l .)"
+
+.PHONY: lint
+lint:
+	LINT_INPUT="$(shell go list ./...)"; golint -set_exit_status $$LINT_INPUT
+
+.PHONY: vet
+vet:
+	VET_INPUT="$(shell go list ./...)"; go vet $$VET_INPUT

--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ build:
 	CGO_ENABLED=0 go build -o $(BINARY)
 
 .PHONY: test
-test: fmt lint vet test-unit
+test: fmt vet test-unit
 
 .PHONY: test-unit
 test-unit:

--- a/README.md
+++ b/README.md
@@ -1,7 +1,9 @@
-# git2consul.go
+[![Build Status](https://travis-ci.com/KohlsTechnology/git2consul-go.svg?branch=master)](https://travis-ci.com/KohlsTechnology/git2consul-go)
 [![Go Report Card](https://goreportcard.com/badge/github.com/KohlsTechnology/git2consul-go)](https://goreportcard.com/report/github.com/KohlsTechnology/git2consul-go)
 
-The git2consul.go tool is used to populate a [Consul](https://www.consul.io) key/value store from a git repo.
+# git2consul-go
+
+The git2consul-go tool is used to populate a [Consul](https://www.consul.io) key/value store from a git repo.
 
 The baseline source code was forked from [go-git2consul](https://github.com/Cimpress-MCP/go-git2consul) which was
 inspired by the orginal [git2consul](https://github.com/breser/git2consul) tool.
@@ -15,7 +17,7 @@ inspired by the orginal [git2consul](https://github.com/breser/git2consul) tool.
 
 ## Installation
 
-git2consul.go comes in two variants:
+git2consul-go comes in two variants:
 * as a single binary file which after downloading can be placed in any working directory - either on the workstation (from which git2consul will be executed) or on the Consul node (depends whether access to the git repository is available from the Consul nodes or not)
 * as a source code that can be build on the user workstation ([How to build from src?](#compiling-from-source))
 
@@ -195,14 +197,7 @@ See [CONTRIBUTING.md](.github/CONTRIBUTING.md) for details.
 
 ### Compiling From Source
 ```
-$ export GO111MODULE=on
-$ go build -o build/bin/git2consul
-```
-
-For Development/Debugging
-```
-$ export GO111MODULE=on
-$ go build -gcflags='-N -l' -o build/bin/git2consul
+$ make build
 ```
 
 ## License
@@ -216,4 +211,3 @@ See [ACKNOWLEDGEMENT.md](ACKNOWLEDGEMENT.md) for details.
 ## Code of Conduct
 
 See [CODE_OF_CONDUCT.md](.github/CODE_OF_CONDUCT.md) for details.
-

--- a/config/load.go
+++ b/config/load.go
@@ -68,7 +68,7 @@ func Load(file string) (*Config, error) {
 	return config, nil
 }
 
-// Check for the validitiy of the configuration file
+// Check for the validity of the configuration file
 func (c *Config) checkConfig() error {
 	for _, repo := range c.Repos {
 		// Check on name

--- a/kv/mocks/kv.go
+++ b/kv/mocks/kv.go
@@ -28,11 +28,13 @@ type item struct {
 	modifyindex uint64
 }
 
+// KV TODO write a useful documentation here
 type KV struct {
 	T     *testing.T
 	items map[string]*item
 }
 
+// Get TODO write a useful documentation here
 func (kv *KV) Get(key string, opts *api.QueryOptions) (*api.KVPair, *api.QueryMeta, error) {
 	kv.T.Logf("KV Get %s", key)
 	if val, ok := kv.items[key]; ok {
@@ -41,6 +43,7 @@ func (kv *KV) Get(key string, opts *api.QueryOptions) (*api.KVPair, *api.QueryMe
 	return nil, nil, nil
 }
 
+// Put TODO write a useful documentation here
 func (kv *KV) Put(kvPair *api.KVPair, wOptions *api.WriteOptions) (*api.WriteMeta, error) {
 	if kv.items == nil {
 		kv.items = make(map[string]*item)
@@ -50,11 +53,13 @@ func (kv *KV) Put(kvPair *api.KVPair, wOptions *api.WriteOptions) (*api.WriteMet
 	return nil, nil
 }
 
+// Delete TODO write a useful documentation here
 func (kv *KV) Delete(key string, wOptions *api.WriteOptions) (*api.WriteMeta, error) {
 	delete(kv.items, key)
 	return nil, nil
 }
 
+// Txn TODO write a useful documentation here
 func (kv *KV) Txn(txnops api.KVTxnOps, opts *api.QueryOptions) (bool, *api.KVTxnResponse, *api.QueryMeta, error) {
 	var checkIndexItem *api.KVTxnOp
 	if length := len(txnops); length > 1 && txnops[length-2].Verb == api.KVCheckIndex {

--- a/kv/mocks/repository.go
+++ b/kv/mocks/repository.go
@@ -27,6 +27,7 @@ import (
 	"gopkg.in/src-d/go-git.v4/storage"
 )
 
+// Repo TODO write a useful documentation here
 type Repo struct {
 	adds   []string
 	Config *config.Repo
@@ -36,27 +37,33 @@ type Repo struct {
 	hashes map[string]plumbing.Hash
 }
 
+// Name TODO write a useful documentation here
 func (r *Repo) Name() string {
 	return "repository_mock"
 }
 
+// GetConfig TODO write a useful documentation here
 func (r *Repo) GetConfig() *config.Repo {
 	return r.Config
 }
 
+// Add TODO write a useful documentation here
 func (r *Repo) Add(path string) {
 	r.adds = append(r.adds, path)
 }
 
+// CheckRef TODO write a useful documentation here
 func (r *Repo) CheckRef(branch string) error {
 	return nil
 }
 
+// CheckoutBranch TODO write a useful documentation here
 func (r *Repo) CheckoutBranch(branch plumbing.ReferenceName) error {
 	r.branch = branch
 	return nil
 }
 
+// DiffStatus TODO write a useful documentation here
 func (r *Repo) DiffStatus(commit string) (object.Changes, error) {
 	var changes object.Changes
 	for _, add := range r.adds {
@@ -66,6 +73,7 @@ func (r *Repo) DiffStatus(commit string) (object.Changes, error) {
 	return changes, nil
 }
 
+// Head TODO write a useful documentation here
 func (r *Repo) Head() (*plumbing.Reference, error) {
 	if r.branch == "" {
 		r.branch = plumbing.NewReferenceFromStrings("master", "").Name()
@@ -74,6 +82,7 @@ func (r *Repo) Head() (*plumbing.Reference, error) {
 	return plumbing.NewHashReference(r.branch, r.hashes[r.branch.Short()]), nil
 }
 
+// Pull TODO write a useful documentation here
 func (r *Repo) Pull(branch string) error {
 	if r.hashes == nil {
 		r.hashes = make(map[string]plumbing.Hash)
@@ -88,22 +97,29 @@ func (r *Repo) Pull(branch string) error {
 	return nil
 }
 
+// ResolveRevision TODO write a useful documentation here
 func (r *Repo) ResolveRevision(plumbing.Revision) (*plumbing.Hash, error) {
 	hash := r.hashes[r.branch.Short()]
 	return &hash, nil
 }
 
+// Worktree TODO write a useful documentation here
 func (r *Repo) Worktree() (*git.Worktree, error) {
 	return &git.Worktree{Filesystem: osfs.New(r.Path)}, nil
 }
 
-func (r *Repo) Lock()   {}
+// Lock TODO write a useful documentation here
+func (r *Repo) Lock() {}
+
+// Unlock TODO write a useful documentation here
 func (r *Repo) Unlock() {}
 
+// GetStorer TODO write a useful documentation here
 func (r *Repo) GetStorer() storage.Storer {
 	return nil
 }
 
+// Branch TODO write a useful documentation here
 func (r *Repo) Branch() plumbing.ReferenceName {
 	return r.branch
 }

--- a/repository/load_test.go
+++ b/repository/load_test.go
@@ -47,7 +47,7 @@ func TestLoadRepos(t *testing.T) {
 	assert.Nil(t, err)
 }
 
-//TestLoadReposExistingDir tests load to exsiting repo.
+//TestLoadReposExistingDir tests load to existing repo.
 func TestLoadReposExistingDir(t *testing.T) {
 	bareDir, err := ioutil.TempDir("", "bare-dir")
 	defer os.RemoveAll(bareDir)

--- a/repository/mocks/repository.go
+++ b/repository/mocks/repository.go
@@ -27,6 +27,7 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing/object"
 )
 
+// InitRemote TODO write useful documentation here
 func InitRemote(t *testing.T) (*git.Repository, string) {
 	repoPath, err := ioutil.TempDir("", "git2consul-remote")
 	if err != nil {
@@ -45,6 +46,7 @@ func InitRemote(t *testing.T) (*git.Repository, string) {
 	return repo, repoPath
 }
 
+// Add TODO write useful documentation here
 func Add(t *testing.T, repo *git.Repository, path string, content []byte) {
 	w, err := repo.Worktree()
 	if err != nil {
@@ -82,6 +84,7 @@ func getSignature() *object.Signature {
 	return sig
 }
 
+// Commit TODO write useful documentation here
 func Commit(t *testing.T, repo *git.Repository, message string) {
 	w, err := repo.Worktree()
 	if err != nil {

--- a/version.go
+++ b/version.go
@@ -22,4 +22,4 @@ var (
 )
 
 // Version of the program
-const Version = "0.1.0"
+const Version = "v0.1.1-dev"

